src/discord.cpp

```diff
@@ -69,11 +81,14 @@ Discord::Discord(QObject* parent)
     Discord_RunCallbacks = reinterpret_cast<Discord_RunCallbacksPrototype>(mpLibrary->resolve("Discord_RunCallbacks"));
     Discord_Shutdown = reinterpret_cast<Discord_ShutdownPrototype>(mpLibrary->resolve("Discord_Shutdown"));
 
-    if (!Discord_Initialize || !Discord_UpdatePresence || !Discord_RunCallbacks || !Discord_Shutdown) {
+    if (!mpLibrary->isLoaded() || !Discord_Initialize || !Discord_UpdatePresence || !Discord_RunCallbacks || !Discord_Shutdown) {
         qDebug() << "Could not find Discord library - searched in:";
         for (auto& libraryPath : qApp->libraryPaths()) {
             qDebug() << "    " << libraryPath;
         }
+        if (auto msg = mpLibrary->errorString(); !msg.isEmpty()) {
```

Comment:
:100:

Reply:
Wondered if you were gonna spot that C++17-ism...! :stuck_out_tongue_closed_eyes: